hbase-server/src/main/java/org/apache/hadoop/hbase/tool/CanaryTool.java

```diff
@@ -274,16 +303,50 @@ public void publishReadTiming(String znode, String server, long msTime) {
     private Map<String, LongAdder> perTableReadLatency = new HashMap<>();
     private LongAdder writeLatency = new LongAdder();
     private final Map<String, List<RegionTaskResult>> regionMap = new ConcurrentHashMap<>();
+    private Map<ServerName, LongAdder> perServerFailuresCount = new ConcurrentHashMap<>();
```

Comment:
Declare as ConcurrentMap?

Reply:
> Declare as ConcurrentMap?

I do this for the following reasons：

1.  At the beginning of the sniffing, we will clear the Map
2. Only Failures Server or Failures table will put an element into the map
3. The sniffing process is concurrent with multiple threads

I do n’t know if I ’m doing it right. If not, any suggestions?

Reply:
I mean change `Map<ServerName, LongAddr>` to `ConcurrentMap<ServerName, LongAddr>`, and also for the above `regionMap`. Not a question why you use `ConcurrentHashMap`...

Reply:
> I mean change `Map<ServerName, LongAddr>` to `ConcurrentMap<ServerName, LongAddr>`, and also for the above `regionMap`. Not a question why you use `ConcurrentHashMap`...

Ok. It's done